content/blog/tune-2/index.Rmd

```diff
@@ -0,0 +1,166 @@
+
+We'll simulate some data with a class imbalance: 
+
+```{r}
+#| label: sim-data
+library(tidymodels)
+
+set.seed(296)
+sim_data <- sim_classification(2000, intercept = -12)
+sim_data |> count(class)
+
+# We'll resampling them via 10-fold cross-validation:
+sim_rs <- vfold_cv(sim_data, strata = class)
+```
+
+We define a tailor object that tags the class probability threshold for optimization: 
+
+```{r}
+#| label: tailor
+tlr_spec <- 
+  tailor() |> 
+  adjust_probability_threshold(threshold = tune())
+```
+
+and also specify a random forest that uses its default tuning parameters: 
+
+```{r}
+#| label: rf
+#| fig-width: 5
+#| fig-height: 7
+#| out-width: 100%
+rf_spec <- rand_forest(mode = "classification")
+rf_thrsh_wflow <- workflow(class ~ ., rf_spec, tlr_spec)
+rf_thrsh_wflow
+```
+
+We can see that we can improve sensitivity by _reducing_ the threshold. The rate of decay in specificity is slow compared to the gain in sensitivity until thresholds less than 10% are used. The Brier score is constant over the threshold since it only uses the estimated class probabilities, which are unaffected by the threshold. 
+
+With a class imbalance, the default 50% threshold yields high specificity but low sensitivity. When we alter the threshold, those numbers will change, and we can select the best trade-off for our application. Let's tune the workflow: 
+
+```{r}
+#| label: tuning
+cls_mtr <- metric_set(roc_auc, sensitivity, specificity)
+
+# To run all resamples in parallel:
+mirai::daemons(10)
+
+set.seed(985)
+rf_thrsh_res <- 
+  rf_thrsh_wflow |> 
+  tune_grid(
+    resamples = sim_rs,
+    grid = tibble(threshold = seq(0, 0.6, by = 0.01)),
+    metrics = cls_mtr
+  )
+```
+
+Let's visualize the results: 
+
+```{r}
+#| label: autoplot
+autoplot(rf_thrsh_res) + lims(y = 0:1)
+```
+
+We can see that we can improve sensitivity by _reducing_ the threshold. The rate of decay in specificity is slow compared to the gain in sensitivity until thresholds less than 10% are used. The Brier score is constant over the threshold since it only uses the estimated class probabilities, which are unaffected by the threshold. 
+
+We've taken great pains to avoid redundant calculations. In this example, for each resample, a single random forest model is trained, and then the postprocessing grid is evaluated. This _conditional execution_ strategy is already used to fit the fewest possible preprocessors, models, and postprocessors. 
+
+For this classification example, recent updates to the [desirability2](https://desirability2.tidymodels.org/#using-with-the-tune-package) package can enable you to jointly find the best sensitivity/specificity trade-off using the threshold parameter _and_ model calibration/separation using other parameters. 
+
+After posit::conf (2025), we'll add more examples and tutorials to tidymodels.org to showcase what we can do with postprocessing. 
```

Comment:
since it is past conf now

```suggestion
We'll add more examples and tutorials to tidymodels.org to showcase what we can do with postprocessing. 
```